tests/conftest.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+import urllib3
+import subprocess
+import time
+
+
+class MockServer:
+    def __init__(self):
+        self.server = subprocess.Popen(
+            ['moogle', '-seclimit', '100', '-elmlimit', '1000'],
+            # stdout=subprocess.PIPE,
+            # stderr=subprocess.DEVNULL
+        )
+        # should be very quick to start up
+        time.sleep(1)
```

Comment:
subprocess.Popen is non-blocking. It will continue immediately.
the other subprocess methods that block will block execution until the subprocess terminates.

A way to handle this would be to implement a ready callback. sleeping for 1 second is simpler for now. The process starts in a few milliseconds so it should be ready always 1 second later.